localgov_base.theme

```diff
@@ -46,13 +46,52 @@ function localgov_base_form_system_theme_settings_alter(&$form, FormStateInterfa
     '#default_value' => theme_get_setting('localgov_base_add_draft_note_to_unpublished_content'),
     '#description'   => t('This adds the word "Draft" to the title of any unpublished content. This is useful if you have removed the pink background from unpublished content.'),
   ];
-
   $form['localgov_base_show_back_to_top_link'] = [
     '#type'          => 'checkbox',
     '#title'         => t('Display a "Back to Top" link on long pages.'),
     '#default_value' => theme_get_setting('localgov_base_show_back_to_top_link'),
     '#description'   => t('This adds a link to the beginning of the page content (<code>#main-content</code>).'),
   ];
+  $form['localgov_base_header_behaviour'] = [
+    '#type' => 'radios',
+    '#title' => t('Header behaviour'),
+    '#default_value' => theme_get_setting('localgov_base_header_behaviour'),
+    '#options' => [
+      'default' => t('Default - scrolls away with the page'),
+      'sticky' => t('Sticky - remains at the top of the page'),
+      'appears_on_scroll' => t('Scroll - appears when scrolling up the page'),
+    ],
+    '#description' => t('Select how you want the header to behave. This setting only apply to anonymous users.'),
+  ];
+
+}
+
+/**
+ * Implements hook_preprocess_html().
+ */
+function localgov_base_preprocess_html(&$variables) {
+  // Add the 'sticky-header' library if the sticky header setting is enabled.
+  if (theme_get_setting('localgov_base_header_behaviour') !== 'default') {
```

Comment:
Here, and above where it's used, in addition to telling people about the dependency, trying to enable it in both the profiles, we could also backstop for it not being there yet with:-

(theme_get_setting('localgov_base_header_behaviour') ?? 'default') !== 'default'

theme_get_setting returns NULL if the setting doesn't exist according to the docs https://api.drupal.org/api/drupal/core%21includes%21theme.inc/function/theme_get_setting/11.x

Reply:
This is safe I've tested it. It just evaluates `NULL !== 'default'` if it's not set.